api-gateway-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/apigateway/OrderServiceProxyController.java

```diff
@@ -0,0 +1,60 @@
+package io.eventuate.examples.tram.ordersandcustomers.apigateway;
+
+import io.eventuate.examples.tram.ordersandcustomers.common.IdGenerator;
+import io.eventuate.examples.tram.ordersandcustomers.orders.domain.events.CreateOrderSagaStartedEvent;
+import io.eventuate.examples.tram.ordersandcustomers.orders.domain.events.OrderDetails;
+import io.eventuate.examples.tram.ordersandcustomers.orders.webapi.CreateOrderRequest;
+import io.eventuate.examples.tram.ordersandcustomers.orders.webapi.CreateOrderResponse;
+import io.eventuate.tram.spring.events.publisher.ReactiveDomainEventPublisher;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.reactive.TransactionalOperator;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Mono;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+
+@RestController
+public class OrderServiceProxyController {
+  private ReactiveDomainEventPublisher domainEventPublisher;
+  private TransactionalOperator transactionalOperator;
+
+
+  @Autowired
+  public OrderServiceProxyController(ReactiveDomainEventPublisher domainEventPublisher, TransactionalOperator transactionalOperator) {
+    this.domainEventPublisher = domainEventPublisher;
+    this.transactionalOperator = transactionalOperator;
+  }
+
+  private ConcurrentHashMap<String, CompletableFuture<CreateOrderResponse>> sagaIdToCompletableFuture = new ConcurrentHashMap<>();
```

Comment:
`sagaIdToCompletableFuture` should be moved to OrderEventConsumer

OrderServiceProxyController should 

```
    CompletableFuture<CreateOrderResponse> response = orderEventConsumer.getResponse(orderAndSagaId)
```

Reply:
ok